src/main/java/com/ipiecoles/java/java350/model/Employe.java

```diff
@@ -58,23 +60,52 @@ public Integer getNbConges() {
         return Entreprise.NB_CONGES_BASE + this.getNombreAnneeAnciennete();
     }
 
-    public Integer getNbRtt(){
+    public Integer getNbRtt() {
         return getNbRtt(LocalDate.now());
     }
 
-    public Integer getNbRtt(LocalDate d){
-        int i1 = d.isLeapYear() ? 365 : 366;int var = 104;
-        switch (LocalDate.of(d.getYear(),1,1).getDayOfWeek()){
-        case THURSDAY: if(d.isLeapYear()) var =  var + 1; break;
-        case FRIDAY:
-        if(d.isLeapYear()) var =  var + 2;
-        else var =  var + 1;
-case SATURDAY:var = var + 1;
-                    break;
+    /**
+     * Méthode permettant de calculer le nombre de jours de RTT dans l'année (au prorata du temps d'activité) selon la formule :
+     * Nb jours RTT =
+     * Nombre de jours dans l'année
+     * - Nombre de jours travaillés dans l'année en plein temps
+     * - Nombre de samedi et dimanche dans l'année
+     * - Nombre de jours fériés ne tombant pas le week-end
+     * - Nombre de congés payés
+     *
+     * @param dateReference la date à laquelle on va calculer le nombre de RTT pour l'année.
+     * @return Nombre de jours de RTT pour l'employé l'année de la date de référence au prorata du temps d'activité.
+     */
+    public Integer getNbRtt(LocalDate dateReference) {
+        int nbJoursAnnee = dateReference.isLeapYear() ? 366 : 365;
+        int nbSamediDimanche = 104;
+        switch (LocalDate.of(dateReference.getYear(), 1, 1).getDayOfWeek()) {
+            case THURSDAY:
+                if (dateReference.isLeapYear())
+                    nbSamediDimanche = nbSamediDimanche + 1; // Cas 1
```

Comment:
Ce n'est pas obligatoire mais il est préférable de mettre les accolades même lorsqu'il n'y a qu'une instruction afin d'éviter les problèmes (ajout d'une instruction dans le if en oubliant de mettre les {}) et améliorer la lisibilité
```suggestion
                if (dateReference.isLeapYear()) {
                    nbSamediDimanche = nbSamediDimanche + 1; // Cas 1
                }
```